File: GuildedRose/src/com/gildedrose/items/RegularItem.java
package com.gildedrose.items;

public class RegularItem extends ItemDecorator {
	
	// Constructors
	protected RegularItem(Item itemToDecorate)
	{
		super(itemToDecorate);
	}

	// Protected methods
	@Override
	protected void updateQuality()
	{
		super.decreaseQualityOnce();
	}
}

File: GildedRose.Net/Items/RegularItem.cs
namespace GildedRose.Items
{
    public class RegularItem : ItemDecorator
    {
    
        // Constructors
        internal RegularItem(Item itemToDecorate) : base(itemToDecorate) { }
    
        // Protected methods
        protected override void UpdateQuality()
        {
            if (this.SellInEnded)
            {
                this.DecreaseQualityTwice();
            }
            else
            {
                this.DecreaseQualityOnce();
            }
        }
        
    }
}